udata/core/user/api.py

```diff
@@ -226,51 +227,57 @@ def get(self):
         search_parser.parse_args()
         return search.query(UserSearch, **multi_to_dict(request.args))
 
-    @api.secure
+    @api.secure(admin_permission)
     @api.doc('create_user')
     @api.expect(user_fields)
     @api.marshal_with(user_fields, code=201)
     @api.response(400, 'Validation error')
     def post(self):
         '''Create a new object'''
-        form = api.validate(UserProfileForm)
+        form = api.validate(UserProfileAdminForm)
         user = form.save()
         return user, 201
 
 
 @ns.route('/<user:user>/', endpoint='user')
 @api.response(404, 'User not found')
-@api.response(410, 'User has been deleted')
+@api.response(410, 'User is not active or has been deleted')
 class UserAPI(API):
+
+    def check_permissions(self, user, readonly=False):
+        if not user.visible:
+            api.abort(410, 'User is not active')
```

Comment:
It cannot be user.active is already a field in the database.
'visible' : means the user is inactive and you are not admin. But when the error is raised, the message must be 'user is not active'. It should not be 'user is not active and you are not admin'.